src/braket/ocean_plugin/braket_sampler.py

```diff
@@ -121,6 +122,27 @@ def edgelist(self) -> Tuple[Tuple[int, int]]:
             sorted(set((u, v) if u < v else (v, u) for u, v in self.properties["couplers"]))
         )
 
+    @property
+    @lru_cache(maxsize=1)
+    def access_optimized_nodelist(self) -> Set[int]:
+        """Set[int]: Set of active qubits for the solver.
+
+        Returning a set allows for near constant existence checks.
+        """
+        return set(self.properties["qubits"])
+
+    @property
+    @lru_cache(maxsize=1)
+    def access_optimized_edgelist(self) -> Dict[int, Set[int]]:
+        """Dict[int, Set[int]]: Dict of active couplers for the solver.
+
+        Returing a dict allows for near constant existence checks.
+        """
+        edges = defaultdict(set)
+        for (u, v) in self.properties["couplers"]:
+            edges[u].add(v)
+        return edges
```

Comment:
I _think_ you can use FrozenDict here because of the SDK's dependency on boltons. Also, each of the values should be frozen.